pkg/scheduler/scheduler.go

```diff
@@ -100,7 +101,12 @@ func NewScheduler(cfg Config, limits Limits, log log.Logger, registerer promethe
 		Name: "cortex_query_scheduler_queue_length",
 		Help: "Number of queries in the queue.",
 	}, []string{"user"})
-	s.requestQueue = queue.NewRequestQueue(cfg.MaxOutstandingPerTenant, s.queueLength)
+
+	s.discardedQueries = promauto.With(registerer).NewCounterVec(prometheus.CounterOpts{
```

Comment:
Please cleanup this metric in `cleanupMetricsForInactiveUser` and add it to tests to verify that.

Reply:
👍